### ISSUE TITLE
[clang] error: attribute 'nothrow' cannot be applied to 'function'

### ISSUE THREAD

**Issue by mikea** [llvm/llvm-project#191487]
For some reason pragma attribute push for nothrow doesn't work.

```c
#pragma clang attribute push(__attribute__((nothrow)), apply_to=function)

int foo() {
    return 42;
}

#pragma clang attribute pop
```

https://godbolt.org/z/s6j8xe5Ej

```
<source>:1:15: error: attribute 'nothrow' cannot be applied to 'function'
    1 | #pragma clang attribute push(__attribute__((nothrow)), apply_to=function)
      |               ^                                                 ~~~~~~~~
<source>:1:45: warning: unused attribute 'nothrow' in '#pragma clang attribute push' region [-Wpragma-clang-attribute]
    1 | #pragma clang attribute push(__attribute__((nothrow)), apply_to=function)
      |                                             ^
<source>:7:15: note: '#pragma clang attribute push' regions ends here
    7 | #pragma clang attribute pop
      |               ^
1 warning and 1 error generated.
Compiler returned: 1
```

**Comment by shafik** [comment]
I spent a little bit of time but could not find what was triggering this specific issue. 

CC @erichkeane

**Comment by AaronBallman** [comment]
I think it's hinted at in the comments here:
```
  // Set to true if this attribute can be used with '#pragma clang attribute'.
  // By default, an attribute is supported by the '#pragma clang attribute'
  // only when:
  // - It has a subject list whose subjects can be represented using subject
  //   match rules.
  // - It has GNU/CXX11 spelling and doesn't require delayed parsing.
  bit PragmaAttributeSupport;
```
because: https://github.com/llvm/llvm-project/blob/7b4c9bb2069536e0df18597795b858e18a1cbaaf/clang/include/clang/Basic/Attr.td#L2784

The subject list is `FunctionLike` which I don't think is supported. Other `FunctionLike` attributes show the same behavior: https://godbolt.org/z/qWWGW8G3v

Note, our documentation claims the attribute is supported in the pragma form: https://clang.llvm.org/docs/AttributeReference.html#nothrow

**Comment by mikea** [comment]
I was also hoping that `nothrow` would be applied to all methods/constructors etc if possible. I guess this `FunctionLike` is what is responsible for this subject?

**Comment by AaronBallman** [comment]
> I was also hoping that `nothrow` would be applied to all methods/constructors etc if possible. I guess this `FunctionLike` is what is responsible for this subject?

Correct, that basically sets the subjects to function declarations and function pointer types. Constructors are already supported: https://godbolt.org/z/MEarGaq1M
